### ISSUE TITLE
Suppressing or Hiding notes and/or lyrics

### ISSUE THREAD

**Issue by AlbertHart** [opensheetmusicdisplay/opensheetmusicdisplay#602]
I am trying to create notation like this for soloing. With Staves and structure - but just chords and no notes.

Is there a way to suppress the display of notes and/or lyrics?

![image](https://user-images.githubusercontent.com/8965352/67541184-45a52b80-f6a5-11e9-95d2-f947ab2c7947.png)

**Comment by sschmidTU** [comment]
Setting `note.PrintObject = false` for all notes should work, because the notes are still rendered, just in transparent color.
The easiest way would probably be to do this:
```js
const transparentColor = "#11111100";
osmd.setOptions({
  defaultColorNotehead: transparentColor
  defaultColorStem: transparentColor
  defaultColorRest: transparentColor
});
osmd.render();
```

**Comment by AlbertHart** [comment]
Setting defaultColor in setOptions() does noit seem to work

I was able to set it in EngravingRules before rendering, but it didn't seem to use my settings.
(I added red and white to see if anything had any effect

```
           const transparentColor = "#00000000"; // last two digits are alpha value
            const red = "#ff000000"; // last two digits are alpha value
            const white = "#ffffff"; // last two digits are alpha value

            osmd.setOptions({
                drawFromMeasureNumber: 1,
                drawUpToMeasureNumber: 4,

                
                defaultColorNotehead: transparentColor,
                defaultColorStem: transparentColor,
                defaultColorRest: transparentColor,

            });

   
            console.log("defaultColorNoteHead: %s", osmd.EngravingRules.defaultColorNotehead);
            osmd.EngravingRules.defaultColorNotehead = red;
            console.log("defaultColorNoteHead: %s", osmd.EngravingRules.defaultColorNotehead);
            
            console.log("defaultColorStem: %s", osmd.EngravingRules.defaultColorStem);
            osmd.EngravingRules.defaultColorStem = white;
            console.log("defaultColorStem: %s", osmd.EngravingRules.defaultColorStem);
```

**Comment by sschmidTU** [comment]
ah, you may have to set the default color options before loading the sample, because the change is not adapted for every (re-)render right now.
here's bars 5-6 of the christmas piece with all default colors transparent:
![image](https://user-images.githubusercontent.com/33069673/67576976-e47e6600-f73f-11e9-932b-65e6dfbef743.png)

By the way, you would set osmd.EngravingRules.DefaultColorNotehead, big "D". the small "d" value is private.

**Comment by AlbertHart** [comment]
I was able to set the values better by getting the 'D's right.

Here is the code I am using. The EngravingRules.DefaultColor stuff is being set properly.

          ```
 const transparentColor = "#00000000"; // last two digits are alpha value
      
            osmd.setOptions({
               
                defaultColorNotehead: transparentColor,
                defaultColorStem: transparentColor,
                defaultColorRest: transparentColor,

            });

            console.log("DefaultColorNoteHead: %s", osmd.EngravingRules.DefaultColorNotehead);
            console.log("DefaultColorStem: %s", osmd.EngravingRules.DefaultColorStem);
            console.log("DefaultColorRest: %s", osmd.EngravingRules.DefaultColorRest);
            console.log("Calling osmd.render()");
            osmd.render();
```

I then discovered that I had to set coloringEnabled: 1, to get it to use these colors.


But I was having a problem setting the stem color properly.
But discovered that if I set:  colorStemsLikeNoteheads: 1,
then they worked.

**Comment by AlbertHart** [comment]
Is there a way to hide the slurs and extension bars?

![image](https://user-images.githubusercontent.com/8965352/67737203-e9594900-f9cf-11e9-8948-8b5acdaa2656.png)

And how about lyrics?

**Comment by sschmidTU** [comment]
There is now a DrawLyrics option: `osmd.setOptions({drawLyrics: false});`

Not drawing notes at all, and thus not drawing slurs and ledger lines, is a bit difficult, because many parts of OSMD rely on notes being present, and creating ghost notes instead in Vexflow can often lead to problems. But i'll take another look, and it should at least be possible to remove some slurs.

**Comment by sschmidTU** [comment]
I also added a DrawSlurs option: `osmd.setOptions({drawSlurs: false});`

With this, the only thing we still need to remove is ledger lines:
![image](https://user-images.githubusercontent.com/33069673/70161725-d9164700-16bc-11ea-9470-d8f90791652b.png)

This could either be done by implementing it in Vexflow (option to set ledger line fill style / stroke style -> transparent), or in OSMD by adding GhostNotes instead of transparent StaveNotes, though as i said this currently creates a lot of problems.

Options used:
```js
osmd.setOptions({drawSlurs: false, drawLyrics: false})
osmd.setOptions({defaultColorStem: "#12345600"})
osmd.setOptions({defaultColorNotehead: "#12345600"})
osmd.setOptions({defaultColorRest: "#12345600"})
// could be aggregated to one setOptions call
```

By the way, it seems like the music notation program you use exports almost every graphical element with color black, which seems excessive. I added a check that if the color is black, OSMD instead uses the default color (which is usually black).
(There are a few cases where this is not what you want (default color is not black, but you want a black note), but they are negligible for now.)
